plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -417,30 +419,36 @@ public boolean accept(File pathname) {
     }
 
     private KubernetesListBuilder generateAppResources(List<ImageConfiguration> images, EnricherManager enricherManager) throws IOException, MojoExecutionException {
-        File[] resourceFiles = KubernetesResourceUtil.listResourceFragments(resourceDir);
-        KubernetesListBuilder builder;
+        try {
+            File[] resourceFiles = KubernetesResourceUtil.listResourceFragments(resourceDir);
+            KubernetesListBuilder builder;
 
-        // Add resource files found in the fabric8 directory
-        if (resourceFiles != null && resourceFiles.length > 0) {
-            log.info("Using resource templates from %s", resourceDir);
-            builder = readResourceFragments(resourceFiles);
-        } else {
-            builder = new KubernetesListBuilder();
-        }
+            // Add resource files found in the fabric8 directory
+            if (resourceFiles != null && resourceFiles.length > 0) {
+                log.info("Using resource templates from %s", resourceDir);
+                builder = readResourceFragments(resourceFiles);
+            } else {
+                builder = new KubernetesListBuilder();
+            }
 
-        // Add locally configured objects
-        if (resources != null) {
-            // TODO: Allow also support resources to be specified via XML
-            addConfiguredResources(builder, images);
-        }
+            // Add locally configured objects
+            if (resources != null) {
+                // TODO: Allow also support resources to be specified via XML
+                addConfiguredResources(builder, images);
+            }
 
-        // Create default resources for app resources only
-        enricherManager.createDefaultResources(builder);
+            // Create default resources for app resources only
+            enricherManager.createDefaultResources(builder);
 
-        // Enrich descriptors
-        enricherManager.enrich(builder);
+            // Enrich descriptors
+            enricherManager.enrich(builder);
 
-        return builder;
+            return builder;
+        } catch (ConstraintViolationException e) {
+            String message = ValidationUtil.createValidationMessage(e.getConstraintViolations());
+            getLog().error("ConstraintViolationException: " + message, e);
```

Comment:
the thing is though without an actual `message` parameter to the exception, nothing at all about any of the violations ever appears in any stack trace - leaving a very frustrated programmer with no clue. Usually stack traces are pretty meaningful - as its common practice to put messages in exceptions as to what went wrong. Passing a message is a nice, polite thing to do - folks can always reformat a better message. But any message is better than none. e.g. just a hello world formatting of the validation exceptions will tell you *exactly* what failed to validate and why in your kubernetes resources; which is pretty damn useful